templates/container.class.tpl

```diff
@@ -45,6 +44,108 @@ class %%CLASSNAME%% extends PluginFieldsAbstractContainerInstance
             }
          }
       }
+
+      /**
+      * Adds the 'entities_id' field to the database table and migrates existing data.
+      *
+      * This block ensures that the 'entities_id' field is created and populated if it
+      * associated item type requires entity assignment
+      */
+      if (getItemForItemtype("%%ITEMTYPE%%")->isEntityAssign() && !$DB->fieldExists($table, 'entities_id')) {
+         $migration->addField($table, 'entities_id', 'fkey', ['after' => 'plugin_fields_containers_id']);
+         $migration->addKey($table, 'entities_id');
+         $migration->executeMigration();
+
+         // migrate data
+         $query = $DB->buildUpdate(
+               $table,
+               ['entities_id' => new QueryParam()],
+               ['id'       => new QueryParam()]
+         );
+         $stmt = $DB->prepare($query);
+
+         //load all entries
+         $data = $DB->request(
+            [
+               'SELECT'    => '*',
+               'FROM'      => $table,
+            ]
+         );
+
+         foreach ($data as $fields) {
+            //load related item
+            $related_item = $DB->request(
+               [
+                  'SELECT'    => '*',
+                  'FROM'      => getTableForItemType($fields['itemtype']),
+                  'WHERE'     => [
+                        'id'      => $fields['items_id'],
+                     ]
+               ]
+            )->current();
+
```

Comment:
It should not happen, but if the related item does not exist anymore you should keep this entry to prevent warnings/errors on `$related_item` access.

```suggestion

            if ($related_item === null) {
                continue;
            }

```